dcrutil/address.go

```diff
@@ -89,6 +75,34 @@ func encodePKAddress(serializedPK []byte, netID [2]byte, algo dcrec.SignatureTyp
 	return base58.CheckEncode(pubKeyBytes, netID)
 }
 
+// AddressParams defines an interface that is used to provide the parameters
+// required when encoding and decoding addresses.  These values are typically
+// well-defined and unique per network.
+type AddressParams interface {
```

Comment:
Just an observation but I feel it's important to make.  As we introduce new address types and/or script version, we should add new address parameter interfaces instead of expanding this one, to reduce breakage.  DecodeAddress can perform a dynamic check for other interface implementations.

Maybe this type should also be graced with a V0 in the name to make this clear?

Reply:
I was considering that as well.  The main reason I didn't was because the new version interfaces would have to also include all of these V0 funcs in order to be eligible to be passed to `DecodeAddress` in order to be able to do a dynamic check which might be a bit annoying.  I'm open to that approach, so long as that annoying aspect is acceptable.

Another option might be to have some type of more generic `AddressParams` interface the function takes that doesn't include the `V0` funcs and then this interface becomes `AddressParamsV0` and embeds the more generic one.  We'd need to come up with a decent name for the function in the generic interface which would basically just serve as a dummy function, then also update `chaincfg.Params` and retag if going with that approach.

Reply:
I see what you mean about having to also implement these V0 methods on, say, future V1 parameter magics.  And in practice, we need to keep those older script types and addresses around forever regardless so we can correctly validate old chains.

So this name is actually fine I think.  We just need to make sure that we start adding V1, V2... interfaces instead of expanding this one and then dynamically check for these in functions which are passed an AddressParams.